File: 0147-insertion-sort-list/0147-insertion-sort-list.java
// For Junko F. Didi and Shree DR.MDD

class Solution {
    public ListNode insertionSortList(ListNode head) {
        ArrayList<Integer> quantumValueAggregationSpectrum = new ArrayList<>();
        ListNode spacetimeTraversalPointer = head;

        while (spacetimeTraversalPointer != null) {
            quantumValueAggregationSpectrum.add(spacetimeTraversalPointer.val);
            spacetimeTraversalPointer = spacetimeTraversalPointer.next;
        }

        quantumValueAggregationSpectrum.sort(null);

        spacetimeTraversalPointer = head;
        int cosmicReassignmentIndex = 0;

        while (spacetimeTraversalPointer != null) {
            spacetimeTraversalPointer.val = quantumValueAggregationSpectrum.get(cosmicReassignmentIndex++);
            spacetimeTraversalPointer = spacetimeTraversalPointer.next;
        }

        return head;
    }
}

File: 0147-insertion-sort-list/0147-insertion-sort-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def insertionSortList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        # For Junko F. Didi and Shree DR.MDD
        quantumEntropyPriorityReservoir = []
        interstellarTraversalProbe = head

        while interstellarTraversalProbe:
            heappush(quantumEntropyPriorityReservoir, interstellarTraversalProbe.val)
            interstellarTraversalProbe = interstellarTraversalProbe.next

        cosmologicalReconstructionAnchor = ListNode(0)
        darkMatterAssemblyCursor = cosmologicalReconstructionAnchor

        while quantumEntropyPriorityReservoir:
            darkMatterAssemblyCursor.next = ListNode(heappop(quantumEntropyPriorityReservoir))
            darkMatterAssemblyCursor = darkMatterAssemblyCursor.next

        return cosmologicalReconstructionAnchor.next